File: Data Structures & Algorithms/subsets-ii/submission-2.cpp
class Solution {
public:

    void DFS(int indx, vector<int>subset,vector<int>& nums, vector<vector<int>>& res)
    {
        res.push_back(subset);
        for(int j = indx;j<nums.size();j++)
        {
            if(j>indx && nums[j] == nums[j-1])
                continue;
            subset.push_back(nums[j]);
            DFS(j+1,subset,nums,res);
            subset.pop_back();
        }
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<vector<int>> res;
        DFS(0,{},nums,res);
        return res;
    }
};

File: Data Structures & Algorithms/web-crawler/submission-2.py
# """
# This is HtmlParser's API interface.
# You should not implement it, or speculate about its implementation
# """
#class HtmlParser(object):
#    def getUrls(self, url):
#        """
#        :type url: str
#        :rtype List[str]
#        """

class Solution:
    def crawl(self, startUrl: str, htmlParser: 'HtmlParser') -> List[str]:
        visited = set()
        res = []
        def dfs(baseurl,url,indx):
            visited.add(url)
            for nexturl in htmlParser.getUrls(url):
                if nexturl not in visited and nexturl[:indx] == baseurl:
                    dfs(baseurl,nexturl,indx)
        indx = min(len(startUrl),len(startUrl.split('/')[2]))
        baseurl = startUrl[0:indx]
        dfs(baseurl,startUrl,indx)
        return list(visited)

        